inc/class-tachyon.php

```diff
@@ -666,6 +669,51 @@ public function filter_srcset_array( $sources, $size_array, $image_src, $image_m
 		return $sources;
 	}
 
+	/**
+	 * Filter meta data to include all image sizes.
+	 *
+	 * This is required to ensure WordPress runs the `wp_calculate_image_srcset`
+	 * filter when calculating the srcset and to ensure new image sizes are
+	 * included when Tachyon filters the srcset.
+	 *
+	 * @param array|bool $data          Array of meta data for the given attachment, or false
+	 *                                  if the object does not exist.
+	 * @param int        $attachment_id Attachment post ID.
+	 * @return array|bool Modified image meta data to include crops not saved on upload.
+	 */
+	public function filter_attachment_meta_data( $data, $attachment_id ) {
+		if ( ! is_array( $data ) || ! wp_attachment_is_image( $attachment_id ) ) {
+			return $data;
+		}
+
+		$image_sizes = self::image_sizes();
+		$mime_type = get_post_mime_type( $attachment_id );
+		$filename = pathinfo( $data['file'], PATHINFO_FILENAME );
+		$ext = pathinfo( $data['file'], PATHINFO_EXTENSION );
+
+		foreach ( $image_sizes as $size => $crop ) {
+			if ( isset( $data['sizes'][ $size ] ) ) {
+				// Meta data is set.
+				continue;
+			}
+
+			if ( 'full' === $size ) {
+				// Full is a special case.
+				continue;
+			}
+
+			// Add meta data with fake WP style file name.
+			$data['sizes'][ $size ] = array(
+				'width' => (int) $crop['width'],
```

Comment:
I've realised since pushing, the dimensions need to be run through `image_resize_dimensions()` rather than left as is. Please don't merge this.